src/include/baselib/jni/JavaVirtualMachine.h

```diff
@@ -247,18 +247,34 @@ namespace bl
 
             static auto instance() -> this_type&
             {
-                BL_MUTEX_GUARD( g_lock );
+                bool instanceCreated = false;
+
+                {
+                    BL_MUTEX_GUARD( g_lock );
+
+                    if( ! g_instance )
+                    {
+                        BL_CHK_T(
+                            true,
+                            g_javaVMDestroyed,
+                            JavaException(),
+                            "JavaVM has already been destroyed"
+                            );
+
+                        g_instance = new JavaVirtualMachineT();
+                        instanceCreated = true;
+                    }
+                }
```

Comment:
possible race condition here... dependencies needs to be changed (discussed offline)